src/main/java/org/folio/service/organization/OrganizationStorageService.java

```diff
@@ -44,28 +45,30 @@ public class OrganizationStorageService extends BaseService implements Organizat
   private AcquisitionsUnitsService acquisitionsUnitsService;
 
   @Override
-  public CompletableFuture<Organization> createOrganization(Organization organization, Context context,
-      Map<String, String> headers) {
+  public Future<Organization> createOrganization(Organization organization, Context context, Map<String, String> headers) {
     logger.debug("createOrganization:: Trying to create organization with name: {}", organization.getName());
-    HttpClientInterface client = getHttpClient(headers);
-    CompletableFuture<Organization> future = new CompletableFuture<>();
+    RequestContext requestContext = new RequestContext(context, headers);
+    Promise<Organization> promise = Promise.promise();
+
     if (isSameAccountNumbers(organization)) {
       logger.warn("crateOrganization:: Account number of organization '{}' is not unique", organization.getName());
-      future.completeExceptionally(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(),
-        ACCOUNT_NUMBER_MUST_BE_UNIQUE.toError()));
-      return future;
+      promise.fail(new CompletionException(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(),
```

Comment:
I think we can use just: promise.fail(new HttpException())

Reply:
fixed